core-bundle/src/Resources/contao/controllers/FrontendIndex.php

```diff
@@ -247,7 +247,7 @@ public function renderPage($pageModel)
 		}
 
 		// Check wether the language matches the root page language
-		if (isset($_GET['language']) && Config::get('addLanguageToUrl') && Input::get('language') != $objPage->rootLanguage)
+		if (isset($_GET['language']) && $objPage->urlPrefix && Input::get('language') != $objPage->rootLanguage)
```

Comment:
I am still not sure about this change in terms of whether it should be `System::getContainer()->getParameter('contao.prepend_locale')` instead of `$objPage->urlPrefix`. The check is only required in legacy routing mode, isn't it? And `$objPage->urlPrefix` is only used in the new routing mode. @aschempp /cc

Reply:
But `$objPage->urlPrefix` gets set to the language if legacy routing is active and prepend_locale is set.

Reply:
I see. Then this is fine. 👍🏻